client/src/partials/primary_cash/income/generic/generic.js

```diff
@@ -1,232 +1,218 @@
-      .catch(function (err) {
-        messenger.error(err);
-      });
-    });
-
-    $scope.formatDebtor = function formatDebtor(debtor) {
-      return [
-        '[' + $scope.projects.get(debtor.project_id).abbr,
-        debtor.reference + ']',
-        debtor.first_name,
-        debtor.last_name
-      ].join(' ');
-    };
-
-    $scope.generate = function generate () {
-      session.receipt.reference_uuid = uuid();
-    };
-
-    $scope.clear = function clear () {
-      session.receipt = {};
-      session.receipt.date = new Date();
-      session.receipt.value = 0.00;
-      session.receipt.cash_box_id = $routeParams.id;
-    };
-
-    function valid () {
-      if (!session || !session.receipt) {
-        session.invalid = true;
-        return;
-      }
-      var r = session.receipt;
-
-      session.invalid = !(isDefined(session.currency) &&
-        isDefined(r.cost) &&
-        r.cost > 0 &&
-        isDefined(r.description) &&
-        isDefined(r.date) &&
-        isDefined(r.cash_box_id));
-    }
+  function update (value) {
+    session.receipt.recipient = value;
+  }
 
-    function update (value) {
-      session.receipt.recipient = value;
-    }
+  function error (err) {
+    messenger.error(err);
+  }
 
-    $scope.$watch('session.receipt', valid, true);
-    $scope.$watch('session.currency', valid, true);
-
-    $scope.submit = function submit () {
-      var data, receipt = session.receipt;
-
-      connect.fetch('/user_session')
-      .then(function (user) {
-
-        data = {
-          uuid          : uuid(),
-          project_id    : $scope.project.id,
-          type          : 'E',
-          date          : util.sqlDate(receipt.date),
-          account_id    : session.ac.id,
-          currency_id   : session.currency.id,
-          cost          : receipt.cost,
-          user_id       : user.id,
-          description   : 'HBB' + '_CP. REC GEN/' + receipt.description, //fix me
-          cash_box_id   : receipt.cash_box_id,
-          origin_id     : 5,
-        };
-
-        return connect.basicPut('primary_cash', [data]);
-      })
-      .then(function () {
-        var item = {
-          uuid              : uuid(),
-          primary_cash_uuid : data.uuid,
-          debit             : data.cost,
-          credit            : 0,
-          document_uuid     : receipt.reference_uuid
-        };
-        return connect.basicPut('primary_cash_item', [item]);
-      })
-      .then(function () {
-        return connect.fetch('/journal/primary_income/' + data.uuid);
-      })
-      .then(function () {
-        // invoice
-        messenger.success($translate.instant('ALLTRANSACTIONS.DATA_POSTED'));
-        $location.path('/invoice/generic_income/' + data.uuid);
-      });
-    };
-
-    function setCurrency (obj) {
-      $scope.session.currency=obj;
-      cache.put('currency', obj);
-    }
+  function submit () {
+    var data, receipt = session.receipt;
+
+    connect.fetch('/user_session')
```

Comment:
Please remove this call.  It downloads the session data available via SessionService.